### DISCUSSION TITLE
Planning operations without material

### DISCUSSION THREAD

**Discussion by etoimene** [frePPLe/frepple#524] [Q&A]
Hi!

Sorry if this is n00b question but ...

I was wondering if I have
list of operations https://frepple.org/docs/current/model-reference/operations.php
no material consumption or production - empty https://frepple.org/docs/current/model-reference/operation-materials.php

resources for operations https://frepple.org/docs/current/model-reference/operation-resources.php
list of resource skills https://frepple.org/docs/current/model-reference/resource-skills.php
list of skills needed for operation

can I get 
operations distributed to resources 
by using frepple solver?

Where would such result be stored?

I am considering using it to calculate resources needed for some project tasks.
Still working on frepple install so probably will have a better picture then.

I just realized there is https://frepple.com/live-demo/ so I'll give it a try.

**Comment by jdetaeye** [comment]
> can I get operations distributed to resources by using frepple solver?

Operations in frepple don't have dates or quantity.  Operations are just defining "producing X takes time A, and uses materials B and requires resources C"

You need as extra information the demand:  "project Y needs to be finished by date D".
Frepple then generates manufacturing orders and purchase orders to plan the project:  "manufacturing order / project task X from date E till date F", "purchase material B by date E", ...

> Where would such result be stored?

The "resource detail" screen is where you will see the assignments of resources to the manufacturing orders / project tasks.

The frepple data model has its roots in discrete manufacturing industries, and not so much for project planning.

**Comment by etoimene** [comment]
Thanks for this clarification. Maybe it would be better to go other route for this. :(

**Reply by binghui1992** [reply]
I'm also a beginner user of FrePPle. I do have same requirements as you did, i.e. to make a plan of a project's tasks which does not require any material nor produce any product, but do require resources like machines or operators. However here are the steps what I achieve the goal by using FrePPle despite of what Mr. De Taeye said "FrePPle is not so much for project plannig": 
1. create a virtual material (e.g. item-none) which will be consumed by all opeartions, and each operation (associated with a project task) will produced another virtual material (e.g. item-operation-1);
2. You could leave the buffer empty or not, you don't care about the virutal material supplier thing, FrePPle will still  make a plan for your project tasks. 

By introducing virtual materials, I think FrePPle is also capable of doing the project tasks planning stuff.

**Reply by jdetaeye** [reply]
We have recently implemented some enhancements to make this modeling easier.   You can now directly define dependencies between operations, which allows you to avoid the intermediate "virtual" items.

See: 
- https://frepple.com/docs/current/examples/operation/operation-dependency 
- https://frepple.com/docs/current/model-reference/operation-dependencies.html 
- https://demo.frepple.com/operation-dependency/supplypath/demand/rocket%20for%202023%20Q2/?noautofilter/

There are some limitations to it (eg doesn't support dependencies between routing operations or between steps of different routings), but it does work pretty well.

Looking forward to contributors and/or sponsors to bring the feature to the next level...
